### DISCUSSION TITLE
Onnx model checker error

### DISCUSSION THREAD

**Discussion by ElMahdiElAnnabi** [Xilinx/finn#513] [Q&A] [answered]
Hi everyone,
when i check the model tfc-w1a2.onnx or any other model with onnx.checker.check_model i receive the following error:
**onnx.onnx_cpp2py_export.checker.ValidationError: No opset import for domain 'finn.custom_op.general'**
any idea??
thanks in advance

**Comment by heborras** [comment, marked as answer]
Hi @ElMahdiElAnnabi,
FINN uses custom node types in it's ONNX graphs to represent functionality not supported by the standard ONNX op-set. This is for example quantization below 8 bits, Multi-threshold activations and so on. Since these nodes are not part of the official op-set the onnx model checker will say that it doesn't know what to do with these nodes. So, the error which you are seeing is in a sense expected behavior, due to how FINN implements its custom ops.

**Comment by ElMahdiElAnnabi** [comment]
hi @HenniOVP, 
thank you so much for your answer!
